Cubical/Foundations/Embedding.agda

```diff
@@ -0,0 +1,81 @@
+{-# OPTIONS --cubical --safe #-}
+
+module Cubical.Foundations.Embedding where
+
+open import Cubical.Core.Everything
+
+open import Cubical.Foundations.Function
+open import Cubical.Foundations.HLevels
+open import Cubical.Foundations.Transport
+open import Cubical.Foundations.Isomorphism
+
+private
+  variable
+    ℓ : Level
+    A B : Set ℓ
+    f : A → B
+    w x : A
+    y z : B
+
+isEmbedding : (A → B) → Set _
+isEmbedding f = ∀ w x → isEquiv {A = w ≡ x} (cong f)
+
+injEmbedding
```

Comment:
Hmm, maybe I'm wrong. But doesn't the hypotheses imply that A is a retract of B ? If that's the case then you don't need `isSet A` as any retract of a set is a set.

Reply:
Ah, okay. I'll try and generalize. I think the HoTT book said it was true "for sets" which was kind of ambiguous.

Reply:
I see, I might very well be wrong as I didn't try to prove it myself! Don't spend too much time on it if you get stuck.

Reply:
I tried to do this, but it seemed impossible. Whichever `isSet` hypothesis I removed seemed to be the one I ended up needing to pull back the other `isSet`. So this might not actually be possible.

Reply:
Ok, good to know!

Reply:
Maybe add a comment saying that the isSet assumptions seem necessary with a pointer to where this is defined in the HoTT book?